src/GridInterpolations.jl

```diff
@@ -159,7 +159,11 @@ interpolate(grid::AbstractGrid, data::Matrix, x::AbstractVector) = interpolate(g
 
 function interpolate(grid::AbstractGrid, data::DenseArray, x::AbstractVector)
     index, weight = interpolants(grid, x)
-    dot(data[index], weight)
+    v = 0.0
+    for (i,data_ind) in enumerate(index)
+        v += data[data_ind]*weight[i]
+    end
+    return v
```

Comment:
Would this work?
```suggestion
    return dot(view(data, index), weight)
```

Reply:
@zsunberg Yes, this works too. It just slows down the computation by around 15 ns. If that is acceptable, I can make this change. Let me know.

Before making the change suggested by Dr. Sunberg

![Screenshot from 2023-12-13 20-49-49](https://github.com/sisl/GridInterpolations.jl/assets/57426103/88d32fab-6542-4f70-b257-4a8a92bb8622)

After making the change suggested by Dr. Sunberg

![Screenshot from 2023-12-13 20-49-55](https://github.com/sisl/GridInterpolations.jl/assets/57426103/c52be551-429b-43dd-92ba-ea691fc0cce3)

Reply:
Either one is fine. `dot` is easier to read, but the for loop performs slightly better.